src/modules/flow/iio/nodes.c

```diff
@@ -776,12 +782,24 @@ humidity_open(struct sol_flow_node *node, void *data, const struct sol_flow_node
 }
 
 static bool
-adc_create_channels(struct iio_double_data *mdata, int device_id)
+adc_create_channels(struct iio_double_data *mdata, int device_id, const char* channelname)
 {
+    char channel_name[NAME_MAX];
+    int ret;
+
     mdata->iio_base.device = sol_iio_open(device_id, &mdata->iio_base.config);
     SOL_NULL_CHECK(mdata->iio_base.device, false);
 
-    mdata->channel_val = iio_add_channel(mdata->scale, mdata->offset, "in_voltage0", &mdata->iio_base);
+    if (channelname) {
+        ret = snprintf(channel_name, sizeof(channel_name), "%s", channelname);
```

Comment:
I believe the `channelname` here is directly obtained from json/fbp setting. But I do not see any checking to it. Would it break the sensor node consistency?
For example: a iio combo sensor [imu MPU6050](http://lxr.free-electrons.com/source/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c#L694), this implementation seems to allow user using a gyro sensor node but when filling in `in_accel_x`, user can obtain accelerometer data from gyro sensor node. So the consistency to a sensor node type is no longer there.

Actually iio sysfs interface format are:

```
<in | out>_<channel type name>[index number]_[modifier name]_<data type name>_[extended name]
|            c*              |      o      |       o       |       c        |       o       |

c = compulsory
o = optional
* most of the time compulsory, unless it is used to be shared by all
```
- channel type name could be referred to http://lxr.free-electrons.com/source/drivers/iio/industrialio-core.c#L54
- modifier name could be referred to http://lxr.free-electrons.com/source/drivers/iio/industrialio-core.c#L85
- data type name could be referred to http://lxr.free-electrons.com/source/drivers/iio/industrialio-core.c#L124
- extended name can be anything

My suggestion is to allow user input index number instead of the whole channel name so that sensor node consistency can still be under controlled across whole Soletta.

What do you guys think?